wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,259 @@
+then the `ScrollView`, the content view will be resized to the size of the
+`ScrollView`.
+The scrollview supports keyboard UI and mousewheel.
+It extends `BaseView`.
+
+Process: **Main**
+
+### Static Methods
+
+#### `ScrollView.fromId(id)`
+
+Returns `ScrollView | null` - The scroll view with the given `id`.
+
+### Instance Methods
+
+#### `scrollView.setContentView(contents)`
+
+Set the contents. The contents is the view that needs to scroll.
+
+#### `scrollView.getContentView()`
+
+Returns `BaseView` - The contents of the `scrollView`.
+
+#### `scrollView.setContentSize(size)`
+
+Set the size of the contents.
+
+#### `scrollView.getContentSize()`
+
+Returns the `size` of the contents.
+
+#### `scrollView.setHorizontalScrollBarMode(mode)`
+
+* `mode` string - Can be one of the following values: `disabled`, `hidden-but-enabled`, `enabled`. Default is `enabled`.
+
+Controls how the horizontal scroll bar appears and functions.
+* `disable` - The scrollbar is hidden, and the pane will not respond to e.g. mousewheel events even if the contents are larger than the viewport.
+* `hidden-but-enabled` - The scrollbar is hidden whether or not the contents are larger than the viewport, but the pane will respond to scroll events.
+*`enabled` - The scrollbar will be visible if the contents are larger than the viewport and the pane will respond to scroll events.
+
+#### `scrollView.setVerticalScrollBarMode(mode)`
+
+Controls how the vertical scroll bar appears and functions.
+
+#### `getHorizontalScrollBarMode()`
+
+Returns `string` - horizontal scrollbar mode.
+
+#### `scrollView.getVerticalScrollBarMode()`
+
+Returns `string` - vertical scrollbar mode.
+
+#### `scrollView.setHorizontalScrollElasticity(elasticity)` _macOS_
+
+* `elasticity` string - Can be one of the following values: `automatic`, `none`, `allowed`. Default is `automatic`.
+
+The scroll view’s horizontal scrolling elasticity mode.
+A scroll view can scroll its contents past its bounds to achieve an elastic effect. 
+When set to `automatic`, scrolling the horizontal axis beyond its document
+bounds only occurs if the document width is greater than the view width,
+or the vertical scroller is hidden and the horizontal scroller is visible.
+* `automatic` - Automatically determine whether to allow elasticity on this axis.
+* `none` - Disallow scrolling beyond document bounds on this axis.
+*`allowed` - Allow content to be scrolled past its bounds on this axis in an elastic fashion.
+
+#### `scrollView.setVerticalScrollElasticity(elasticity)` _macOS_
+
+The scroll view’s vertical scrolling elasticity mode.
+
+#### `scrollView.getHorizontalScrollElasticity()` _macOS_
+
+Returns `string` - The scroll view’s horizontal scrolling elasticity mode.
+
+#### `view.getVerticalScrollElasticity()` _macOS_
+
+Returns `string` - The scroll view’s vertical scrolling elasticity mode.
+
+#### `scrollView.scrollTo(point)` _macOS_
+
+* `point` - The point in the `contentView` to scroll to.
```

Comment:
perhaps a way to get the current scroll position would be useful?

Reply:
That's right. Actually I defined the functions: setScrollPosition, getScrollPosition, getMaximumScrollPosition. I'm adjusting this spec.

Reply:
Is this need on Win / Linux, and do you have plans in that area?  This API seems pretty elaborate for single-platform use.

Reply:
There is a lot of `_macOS_`-only API here and some of it, e.g. `view.setScrollPosition()`, seems important to having a well-behaved app. Is this functionality needed for Windows and Linux as well?

If I'm misunderstanding, e.g. if this is only relevant on macOS, maybe the docs could have sample code of what a well-behaved client that runs on all three would look like.

Reply:
That's right, this functionality should exist on all platforms. I have not yet defined how consistent behavior would be achieved, Chromium has View::ScrollRectToVisible defined. I will focus on that in the next step.

Reply:
fixed

Reply:
moved from _macOS_ to _Experimental_ in 372905c240d9fc6953d2e7aefca0b1d6fcd2b795

Reply:
[]().```RAD.E¥E.S5¥```.